### PR TITLE
bpo-45847: Port _lfprof, _opcode, _asyncio, _queue, _statistics, and _typing to PY_STDLIB_MOD_SIMPLE

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue45847

Automerge-Triggered-By: GH:tiran

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -6076,15 +6076,21 @@ PY_STDLIB_MOD_SIMPLE([time], [], [$TIMEMODULE_LIB])
...
 PY_STDLIB_MOD_SIMPLE([_datetime], [], [$TIMEMODULE_LIB $LIBM])
 PY_STDLIB_MOD_SIMPLE([_heapq])
 PY_STDLIB_MOD_SIMPLE([_json])
+PY_STDLIB_MOD_SIMPLE([_lsprof])
 PY_STDLIB_MOD_SIMPLE([math], [], [$LIBM])
+PY_STDLIB_MOD_SIMPLE([_opcode])
 PY_STDLIB_MOD_SIMPLE([_pickle])
+PY_STDLIB_MOD_SIMPLE([_queue])
 PY_STDLIB_MOD_SIMPLE([_random])
+PY_STDLIB_MOD_SIMPLE([_statistics])
```
Comment (tiran) [inline configure.ac]: ``Modules/_statisticsmodule.c`` uses ``fabs()`` and therefore requires ``$LIBM``, too.

Comment (erlend-aasland) [reply configure.ac]: Thanks for catching that!

Comment (erlend-aasland) [reply configure.ac]: Resolved in 4b6fb76eaa1fa7214661eeedd355f6924a10dbeb. I also took the liberty to group the libm dependent modules in that commit. See if that looks ok to you.

File: setup.py [hunk 2]
```diff
@@ -1015,19 +1015,15 @@ def detect_simple_extensions(self):
...
-        # _queue module
-        self.add(Extension("_queue", ["_queuemodule.c"]))
-        # _statistics module
-        self.add(Extension("_statistics", ["_statisticsmodule.c"]))
-        # _typing module
-        self.add(Extension("_typing", ["_typingmodule.c"]))
+        self.addext(Extension("_asyncio", ["_asynciomodule.c"]))
+        self.addext(Extension("_queue", ["_queuemodule.c"]))
+        self.addext(Extension("_statistics", ["_statisticsmodule.c"]))
+        self.addext(Extension("_typing", ["_typingmodule.c"]))
```
Comment (tiran) [inline setup.py]: You removed some of the comments. Now it looks like all four C extensions are asyncio speedups.

**Review by tiran — APPROVED** [review]

**Pull request merged**